lib/flow.js

```diff
@@ -253,7 +253,7 @@ module.exports.waterfall = function (tasks, callback) {
             called = true;
 
             if (err) {
-                return callback(err);
+                return callback.apply(err, arguments);
```

Comment:
Shouldn't this be `callback.apply(null, err, arguments)`?

Even better since this is ES6, I think you can skip `apply` entirely and do `callback(..., err, arguments)` or something similar.